packages/loader/container-loader/src/container.ts

```diff
@@ -652,10 +652,14 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.storage = new ContainerStorageAdapter(
             () => {
                 if (this.attachState !== AttachState.Attached) {
-                    this.logger.sendErrorEvent({
-                        eventName: "NoRealStorageInDetachedContainer",
-                    });
-                    throw new Error("Real storage calls not allowed in Unattached container");
+                    if (this.loader.services.detachedBlobStorage !== undefined) {
+                        return this.loader.services.detachedBlobStorage as IDocumentStorageService;
```

Comment:
This cast is not safe. I think there has to be a wrapper implementation of IDocumentStorageService that has some default implementation for all methods, but also likely log error into telemetry if any of the methods except blob API are called.

Reply:
Instead of letting casting IDetachedBlobStorage to IDocumentStorageService and letting the JavaScript engine throw 'undefined reference' if someone calls one of the non-blob methods, Vlad is asking to wrap IDetachedBlobStorage in a shim that implements IDocumentStorageService and does the following for the non-blob methods:

```ts
                    this.logger.sendErrorEvent({
                        eventName: "NoRealStorageInDetachedContainer",
                    });
                    throw new Error("Real storage calls not allowed in Unattached container");
```